rbx_binary/src/serializer.rs

```diff
@@ -413,6 +427,18 @@ impl<'a, W: Write> BinarySerializer<'a, W> {
                     prop_info.aliases.insert(prop_name.clone());
                 }
             }
+
+            if let Variant::SharedString(shared_string) = prop_value {
+                if !self.shared_string_ids.contains_key(shared_string) {
+                    let id = self
+                        .shared_strings
+                        .len()
+                        .try_into()
+                        .map_err(|_| InnerError::TooManySharedStrings)?;
```

Comment:
I don't think this will ever get anywhere near overflowing in practice. It should be okay to just use `as u32` here, or even `try_into().unwrap()`.